### PR TITLE
Chapter1/homework/4

### PR DESCRIPTION
**Pull request by davinsoman**
I see my mistake with the pull request, sorry about that

### CODE REVIEW

**Comment by davinsoman** [comment]
Alright will see what I need to fix for this to work rn

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        }
+
+        public static float PromptFloat(string message)
+        {
+            Console.WriteLine(message);
+            var GrabFloat = Console.ReadLine();
+            //Checks if input of age is empty or not
+            var FloatNumberCheck = string.IsNullOrWhiteSpace(GrabFloat);
+            //Checks if input is a valid number or not
+            if (FloatNumberCheck != true)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Codacy Issue found: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4913227)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 7
- Added 4
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson4.cs  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4913227&bid=16263467)

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        }
+
+        public static int PromptInt(string message)
+        {
+            Console.WriteLine(message);
+            var GrabInt = Console.ReadLine();
+            //Checks if input of age is empty or not
+            var IntNumberCheck = string.IsNullOrWhiteSpace(GrabInt);
+            //Checks if input is a valid number or not
+            if (IntNumberCheck != true)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Codacy Issue found: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4913227)

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        {
+            Console.WriteLine(message);
+            var GrabFloat = Console.ReadLine();
+            //Checks if input of age is empty or not
+            var FloatNumberCheck = string.IsNullOrWhiteSpace(GrabFloat);
+            //Checks if input is a valid number or not
+            if (FloatNumberCheck != true)
+            {
+                bool CheckValidNumber = int.TryParse(GrabFloat, out int measurement);
+                if (CheckValidNumber == true)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Codacy Issue found: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4913227)

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+            while (true);
+
+        }
+
+        public static string PromptString(string message)
+        {
+            Console.WriteLine(message);
+            var GrabString = Console.ReadLine();
+            var StringCheck = string.IsNullOrWhiteSpace(GrabString);
+            if (StringCheck == true)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Codacy Issue found: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4913227)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 7
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson4.cs  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4913227&bid=16263467)

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+            }
+            while (true);
+
+        }
+
+        public static string PromptString(string message)
+        {
+            Console.WriteLine(message);
+            var GrabString = Console.ReadLine();
+            var StringCheck = string.IsNullOrWhiteSpace(GrabString);
```
Comment (ellmaster) [inline Src/BootCamp.Chapter/Lesson4.cs]: Those names should begin with a lowercase letter.

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+                return GrabString;
+            }
+        }
+
+        public static int PromptInt(string message)
+        {
+            Console.WriteLine(message);
+            var GrabInt = Console.ReadLine();
+            //Checks if input of age is empty or not
+            var IntNumberCheck = string.IsNullOrWhiteSpace(GrabInt);
```
Comment (ellmaster) [inline Src/BootCamp.Chapter/Lesson4.cs]: Same here.

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+                return 0;
+            }
+        }
+
+        public static float PromptFloat(string message)
+        {
+            Console.WriteLine(message);
+            var GrabFloat = Console.ReadLine();
+            //Checks if input of age is empty or not
+            var FloatNumberCheck = string.IsNullOrWhiteSpace(GrabFloat);
```
Comment (ellmaster) [inline Src/BootCamp.Chapter/Lesson4.cs]: Same here.

File: Src/BootCamp.Chapter/Checks.cs [hunk 8]
```diff
@@ -16,25 +12,25 @@ public static class Checks
     {
         public static int PromptInt(string message)
         {
-            // To do: call your implementation. 
+            Lesson4.PromptInt(message);
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Checks.cs]: You just call a function, but you need to return it too.

File: Src/BootCamp.Chapter/Checks.cs [hunk 9]
```diff
@@ -16,25 +12,25 @@ public static class Checks
...
             return 0;
         }
 
         public static string PromptString(string message)
         {
-            // To do: call your implementation. 
+            Lesson4.PromptString(message);
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Checks.cs]: You just call a function, but you need to return it too.

File: Src/BootCamp.Chapter/Checks.cs [hunk 10]
```diff
@@ -16,25 +12,25 @@ public static class Checks
...
             return "";
         }
 
         public static float PromptFloat(string message)
         {
-            // To do: call your implementation. 
+            Lesson4.PromptFloat(message);
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Checks.cs]: You just call a function, but you need to return it too.

File: Src/BootCamp.Chapter/Checks.cs [hunk 11]
```diff
@@ -16,25 +12,25 @@ public static class Checks
...
             return 0;
         }
 
         public static float CalculateBmi(float weight, float height)
         {
-            // To do: call your implementation. 
+            Lesson4.CalculateBmi(weight, height);
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Checks.cs]: You just call a function, but you need to return it too.

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 7
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson4.cs  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4913227&bid=16263467)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 7
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson4.cs  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4913227&bid=16263467)

File: Src/BootCamp.Chapter/Checks.cs [hunk 12]
```diff
@@ -16,26 +12,26 @@ public static class Checks
...
-            return 0;
+            //Lesson4.PromptFloat(message);
+            return Lesson4.PromptFloat(message);
         }
 
         public static float CalculateBmi(float weight, float height)
         {
-            // To do: call your implementation. 
-            return 0;
+            //Lesson4.CalculateBmi(weight, height);
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Checks.cs]: Codacy Issue found: [Remove this commented out code.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4913227)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 7
- Added 4
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson4.cs  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4913227&bid=16263467)

File: Src/BootCamp.Chapter/Checks.cs [hunk 13]
```diff
@@ -16,26 +12,26 @@ public static class Checks
...
-            return 0;
+            //Lesson4.PromptInt(message);
+            return Lesson4.PromptInt(message);
         }
 
         public static string PromptString(string message)
         {
-            // To do: call your implementation. 
-            return "";
+            //Lesson4.PromptString(message);
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Checks.cs]: Codacy Issue found: [Remove this commented out code.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4913227)

File: Src/BootCamp.Chapter/Checks.cs [hunk 14]
```diff
@@ -16,26 +12,26 @@ public static class Checks
...
+            return Lesson4.PromptString(message);
         }
 
         public static float PromptFloat(string message)
         {
-            // To do: call your implementation. 
-            return 0;
+            //Lesson4.PromptFloat(message);
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Checks.cs]: Codacy Issue found: [Remove this commented out code.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4913227)

File: Src/BootCamp.Chapter/Checks.cs [hunk 15]
```diff
@@ -16,26 +12,26 @@ public static class Checks
     {
         public static int PromptInt(string message)
         {
-            // To do: call your implementation. 
-            return 0;
+            //Lesson4.PromptInt(message);
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Checks.cs]: Codacy Issue found: [Remove this commented out code.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4913227)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 7
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson4.cs  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4913227&bid=16263467)

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 16]
```diff
@@ -0,0 +1,142 @@
...
+namespace BootCamp.Chapter
+{
+    public class Lesson4
+    {
+        public static void Demo()
+        {
+            Console.WriteLine("Please enter your data" + Environment.NewLine);
+            do
+            {
+                var name = PromptString("What is your first name?");
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Block in do while could be a few functions like
```
ReadPersonInfo()
DisplayPersonInfo()
```

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 17]
```diff
@@ -0,0 +1,142 @@
...
+                Console.WriteLine("Weight: " + weight + "kg");
+                Console.WriteLine("Height: " + heightCm / 100 + "m");
+                Console.WriteLine("BMI: " + bmi);
+
+
+                Console.WriteLine(Environment.NewLine + "Continue with a new set of inputs? " +
+                "Press Y to continue or press any other button to exit the program" + Environment.NewLine);
+
+                string answer = Console.ReadLine().ToLowerInvariant();
+                if (answer == "y")
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Nice idea for using a loop :)

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 18]
```diff
@@ -0,0 +1,142 @@
...
+            {
+                return grabString;
+            }
+        }
+
+        public static int PromptInt(string message)
+        {
+            Console.WriteLine(message);
+            var grabInt = Console.ReadLine();
+            //Checks if input of age is empty or not
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: The comments are redundant. The code speaks for itself.

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 19]
```diff
@@ -0,0 +1,142 @@
...
+                {
+                    Console.WriteLine("Height cannot be less than zero, but was " + height + ".");
+                }
+
+            }
+
+            return weight / (height * height);
+
+        }
+
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Redundant blank lines at the end.

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 20]
```diff
@@ -0,0 +1,142 @@
...
+                return grabString;
+            }
+        }
+
+        public static int PromptInt(string message)
+        {
+            Console.WriteLine(message);
+            var grabInt = Console.ReadLine();
+            //Checks if input of age is empty or not
+            var intNumberCheck = string.IsNullOrWhiteSpace(grabInt);
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Better name ``isEmpty``

Comment (Almantask) [reply Src/BootCamp.Chapter/Lesson4.cs]: And you can be smarter about this.
```
if(string.IsNullOrEmpty(grabInt) return 0;
```

Comment (Almantask) [reply Src/BootCamp.Chapter/Lesson4.cs]: Then you won't need to have nesting

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 21]
```diff
@@ -0,0 +1,142 @@
...
+            //Checks if input is a valid number or not
+            if (!intNumberCheck)
+            {
+                bool checkValidNumber = int.TryParse(grabInt, out int age);
+                if (checkValidNumber)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: ``isNumber`` is a better name.

**Pull request closed without merging**